dynet/training.cc

```diff
@@ -191,9 +194,15 @@ void MomentumSGDTrainer::update_lookup_params(real gscale, size_t idx) {
   auto & p = model->lookup_parameters_list()[idx];
   update_rule(gscale, {&p->all_values, &p->all_grads, &vlp[idx].all_h});
 }
-void MomentumSGDTrainer::alloc_impl() {
-  vp = allocate_shadow_parameters(*model);
-  vlp = allocate_shadow_lookup_parameters(*model);
+unsigned MomentumSGDTrainer::alloc_impl() {
+  auto v = allocate_shadow_parameters(*model, aux_allocated);
+  vp.insert(vp.end(), v.begin(), v.end());
```

Comment:
There's a bunch of copy-pasted inserts here. It'd be better if we didn't have to copy-paste this every time. Is there any way around this by doing the modification within `allocate_shadow_parameters`, etc?

Reply:
Sure, we could pass a reference to the vector we want to append to. Would that be better?

Reply:
Yep.